engine/src/main/java/org/terasology/persistence/typeHandling/extensionTypes/ComponentClassTypeHandler.java

```diff
@@ -17,22 +17,23 @@
 
 import org.terasology.entitySystem.Component;
 import org.terasology.persistence.typeHandling.StringRepresentationTypeHandler;
+import org.terasology.persistence.typeHandling.TypeHandlerContext;
 
 public class ComponentClassTypeHandler extends StringRepresentationTypeHandler<Class<? extends Component>> {
 
+    TypeHandlerContext context;
+
+    public ComponentClassTypeHandler(final TypeHandlerContext context) {
+        this.context = context;
+    }
+
     @Override
     public String getAsString(Class<? extends Component> item) {
-        return item.getName();
+        return context.getSandbox().getSubTypeIdentifier(item, Component.class);
     }
 
     @Override
     public Class<? extends Component> getFromString(String representation) {
-        try {
-            final Class<? extends Component> clazz = Class.forName(representation).asSubclass(Component.class);
-            return clazz;
-        } catch (ClassNotFoundException e) {
-            e.printStackTrace();
-            return null;
-        }
+        return context.getSandbox().findSubTypeOf(representation, Component.class).orElse(null);
```

Comment:
This is the only thing in this PR that's not just a straight deletion. And it looks okay from here.

It _does_ change the constructor, but presumably that's the sort of thing that you'd find out pretty fast as soon as you tried to compile against it.

Actually ... where _is_ this class used? it's not in my workspace.

Reply:
Ah, this is actually some sneaky little code cleanup :nerd_face:

This class here existed, but was never used. However, there existed another inner class with exact same name (and purpose) in `ComponentClassTypeHandlerFactory`. As I don't like inner classes, I thought it would be better to have the correct code in this class, which is now used by the factory. 

See https://github.com/MovingBlocks/Terasology/pull/4016/files#diff-7fd63f399c356366e5f1abab7dc34ebcL40-L57